include/singa/core/platform.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I put platform declaration in device.h because platform.cc is under src/core/device/.
if you change it, you may also need to change the src/python/swig/core_device.h, and the test/singa/test_platform.cc.
better put it back into device.h for the time being.